httpx/middleware.py

```diff
@@ -138,15 +138,22 @@ def redirect_url(self, request: AsyncRequest, response: AsyncResponse) -> URL:
 
         return url
 
-    def redirect_headers(self, request: AsyncRequest, url: URL) -> Headers:
+    def redirect_headers(self, request: AsyncRequest, url: URL, method: str) -> Headers:
         """
-        Strip Authorization headers when responses are redirected away from
-        the origin.
+        Return the headers that should be used for the redirect request.
         """
         headers = Headers(request.headers)
+
         if url.origin != request.url.origin:
+            # Strip Authorization headers when responses are redirected away from
+            # the origin.
             del headers["Authorization"]
-            del headers["host"]
+            del headers["Host"]
+
+        if method != request.method and method == "GET":
+            # Strip Content-Length headers when we've switch to a 'GET' request.
+            del headers["Content-Length"]
```

Comment:
Should we also be stripping `Transfer-Encoding`? What other headers are only used when a request has a body?

Reply:
Ah good call, yeah.

Those two are at any rate the only *automatic* ones that we need to concern ourselves with. https://github.com/encode/httpx/blob/fa74aaa6dbd9578b094d2dffb2c9f409e2bd8f34/httpx/models.py#L549